### PR TITLE
Removed support of adding extensions on the fly (BC break)

### PR DESCRIPTION
**Pull request by dg**
[Closes #50]

### CODE REVIEW

File: src/DI/Compiler.php [hunk 1]
```diff
@@ -106,14 +106,26 @@ public function processParameters()
...
+			$extension->setConfig(array_diff_key($this->config[$name], self::$reserved));
+		}
+
+		foreach ($extensions as $extension) {
+			$extension->loadConfiguration();
+		}
+
+		if ($extra = array_diff_key($this->extensions, $extensions, $first)) {
+			$extra = implode("', '", array_keys($extra));
+			throw new Nette\DeprecatedException("Extensions '$extra' was added while container is compiled.");
```
Comment (fprochazka) [inline src/DI/Compiler.php]: `was beeing compiled` ?

Comment (fprochazka) [inline src/DI/Compiler.php]: `were added while container was beeing compiled` ?

Comment (dg) [inline src/DI/Compiler.php]: What do you mean?

Comment (JanTvrdik) [inline src/DI/Compiler.php]: was -> were

Comment (JanTvrdik) [inline src/DI/Compiler.php]: was **being** compiled

Comment (dg) [inline src/DI/Compiler.php]: Oh, fixed

File: src/DI/Compiler.php [hunk 2]
```diff
@@ -106,17 +106,28 @@ public function processParameters()
...
 		}
 
-		if ($extra = array_diff_key($this->config, self::$reserved, $this->extensions)) {
+		foreach ($extensions as $extension) {
+			$extension->loadConfiguration();
+		}
+
+		if ($extra = array_diff_key($this->extensions, $extensions, $first)) {
+			$extra = implode("', '", array_keys($extra));
+			throw new Nette\DeprecatedException("Extensions '$extra' was added while container was being compiled.");
```
Comment (fprochazka) [inline src/DI/Compiler.php]: * `Extensions were`, or no?

Comment (JanTvrdik) [inline src/DI/Compiler.php]: yes

**Pull request merged**